opm/simulators/wells/WellConstraints.hpp

```diff
@@ -39,15 +39,15 @@ using RegionId = int;
 class Rates;
 template<class Scalar> class SingleWellState;
 class SummaryState;
-class WellInterfaceGeneric;
+template<class Scalar> class WellInterfaceGeneric;
 enum class WellInjectorCMode;
 enum class WellProducerCMode;
 
 //! \brief Class for computing well group constraints.
 class WellConstraints {
 public:
     //! \brief Constructor sets reference to well.
-    WellConstraints(const WellInterfaceGeneric& well) : well_(well) {}
+    WellConstraints(const WellInterfaceGeneric<double>& well) : well_(well) {}
```

Comment:
Is this to be changed later or should it be done in this PR?

Reply:
will be changed in an upcoming PR where WellConstraints gets its template param.